### ISSUE TITLE
Exposed subsystem API

### ISSUE THREAD

**Issue by gnawinganimal** [SouthEugeneRoboticsTeam/sertain#77]
Recently I noticed that there are some functions that you should always be able to access within a subsystem. Take a look at this subsystem:
```kotlin
class Arm : Subsystem() {
  private val motor = MotorController(Talon(1))

  val pos get() = motor.position

  fun move(output: Double) {
    motor.setPercentOutput(output)
  }
}
```
Here, we need to restrict the usage of `move` to commands, but its not a problem if `pos` is accessed from outside of a command. If fact, it would be extremely convenient. Here is a simple solution:
```kotlin
object Arm : Subsystem {
    private val motor = MotorController(TalonId(0))

    val pos get() = motor.position
    
    class Lock(key: Key<Arm>) {
        fun move(output: Double) {
            motor.setPercentOutput(output)
        }
    }
}
```
Then it can be used in a command:
```kotlin
suspend fun moveArm() = Arm { key ->
  val dt = Arm.Lock(key)
  periodic {
    dt.move(1.0)
  }
}
```
As you can see, functions that can must be restricted can be moved into the `Lock` class, and can only be access with an the subsystem's specific key, which can only be accessed from the `use` function.

**Comment by gnawinganimal** [comment]
We can also expand this `Lock` pattern to motors themselves
```kotlin
object Arm : Subsystem {
    private val motor = motor(TalonId(0))

    val pos get() = motor.position
    
    class Lock(key: Key<Arm>) {
        fun move(output: Double) {
            motor.Lock(key).setPercentOutput(output)
        }
    }
}
```

**Comment by gnawinganimal** [comment]
You could also just take `Key<Arm>` as a function parameter instead of creating a class
```kotlin
object Arm : Subsystem {
    private val motor = motor(TalonId(0))

    val pos get() = motor.position
    
    fun move(key: Key<Arm>, output: Double) {
        motor.Lock(key).setPercentOutput(output)
    }
}
```

**Comment by gnawinganimal** [comment]
Solved by #78
